### ISSUE TITLE
Pip packages with dependencies don't work in bzlmod

### ISSUE THREAD

**Issue by calliecameron** [bazelbuild/rules_python#884]
# 🐞 bug report

### Affected Rule

Bzlmod pip extension.

### Description

Adding a pip package that has dependencies to a module's requirements causes a build error. As far as I can see, the packages used in `examples/bzlmod` don't have dependencies, so don't trigger this error.

Is there a workaround?

## 🔬 Minimal Reproduction

Make the following changes to `examples/bzlmod`:

1. in `requirements.in`: add `pylint~=2.15.5`
2. run `bazel run :requirements.update`
3. in `BUILD.bazel`: add `requirement("pylint")` to the deps of `lib`
4. `bazel build :all`

## 🔥 Exception or Error

```
ERROR: /home/callie/.cache/bazel/_bazel_callie/b170dd4516a10b127f4fcc4b68613d37/external/rules_python~override~pip~pip_pylint/BUILD.bazel:22:11: no such package '@[unknown repo 'pip_typing_extensions' requested from @rules_python~override~pip~pip_pylint]//': The repository '@[unknown repo 'pip_typing_extensions' requested from @rules_python~override~pip~pip_pylint]' could not be resolved: No repository visible as '@pip_typing_extensions' from repository '@rules_python~override~pip~pip_pylint' and referenced by '@rules_python~override~pip~pip_pylint//:pkg'
```

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
Linux Mint 20.1
  </code>
</pre>

**Output of `bazel version`:**

<pre>
  <code>
Bazelisk version: v1.13.2
Build label: 6.0.0rc1
Build target: bazel-out/k8-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Mon Oct 31 10:41:40 2022 (1667212900)
Build timestamp: 1667212900
Build timestamp as int: 1667212900
  </code>
</pre>

**Rules_python version:**

<pre>
  <code>
0.14.0
  </code>
</pre>

**Anything else relevant?**

**Comment by f0rmiga** [comment]
cc @alexeagle @kormide

**Comment by kormide** [comment]
Thanks, I'll look into this.

**Comment by kormide** [comment]
Fixed in https://github.com/bazelbuild/rules_python/pull/893. Will be in the next release.

**Comment by calliecameron** [comment]
This still isn't working fully, due to differences between using rules_python as a local_path_override (as in the example), and pulled from a registry. To reproduce, apply the following patch to commit 0054574 (the extra registry in .bazelrc was set up for testing, and points to the current head of rules_python):

```
diff --git a/examples/bzlmod/.bazelrc b/examples/bzlmod/.bazelrc
index b3a24e8..f4d4a63 100644
--- a/examples/bzlmod/.bazelrc
+++ b/examples/bzlmod/.bazelrc
@@ -1 +1 @@
-common --experimental_enable_bzlmod
+common --experimental_enable_bzlmod --registry=https://raw.githubusercontent.com/calliecameron/markdown-makefile/bazel-test/registry --registry=https://raw.githubusercontent.com/bazelbuild/bazel-central-registry/main
diff --git a/examples/bzlmod/MODULE.bazel b/examples/bzlmod/MODULE.bazel
index 14a338c..0c427db 100644
--- a/examples/bzlmod/MODULE.bazel
+++ b/examples/bzlmod/MODULE.bazel
@@ -4,12 +4,7 @@ module(
     version = "0.0.0",
 )
 
-bazel_dep(name = "rules_python", version = "0.0.0")
-
-local_path_override(
-    module_name = "rules_python",
-    path = "../..",
-)
+bazel_dep(name = "rules_python", version = "0.14.0.1")
 
 python = use_extension("@rules_python//python:extensions.bzl", "python")
 
diff --git a/examples/bzlmod/__init__.py b/examples/bzlmod/__init__.py
index 955eb5a..10698c4 100644
--- a/examples/bzlmod/__init__.py
+++ b/examples/bzlmod/__init__.py
@@ -1,5 +1,7 @@
+import pylint
 from tabulate import tabulate
 
 
 def main(table):
+    pylint.run_pylint(['__init__.py'])
     return tabulate(table)
```

Run `bazel test :all` in examples/bzlmod, and the test fails with an error message about 'problem importing module'. With the same patch, but rules_python used through local_path_override, the test correctly reports some pylint findings.

The only difference I can see between the two cases is the folder name pylint gets put in under runfiles: with a local override, it's `rules_python~override~pip~pip_pylint`, and with a registry it's `rules_python~0.14.0.1~pip~pip_pylint`, i.e. the registry version contains dots, and the override version doesn't.

I don't know if pylint is doing anything unusual here - I've got some other packages working fine with rules_python from a registry, e.g. mypy.

**Comment by alexeagle** [comment]
Yeah that's https://github.com/bazelbuild/bazel/issues/16124 - bzlmod changed the assumption all the runfiles libraries make.
Note that's marked there that rules_python has to do its own thing. We've had discussions about it on slack, I'd like to just publish a library to pypi but someone has to do the work.

**Comment by alexeagle** [comment]
The original issue is resolved now, the runfiles issue is separate. Still working to post 0.15.0 to BCR but nothing left to do on this issue.
